1.3.c

```diff
@@ -1,121 +1,111 @@
 #include <stdio.h>
+#include <string.h>
 #include <stdbool.h>
+// I am did't do maxsize , because my IDE cant scanf this.
 
-#define MAXIMUMSIZE 256
-
-bool isCorrectString(char* string)
+int IncomingDataStatus(char symbol, char state) 
 {
-    int state = 0;
-    char currentCharacter = -1;
-    int counter = 0;
-    bool isContinue = true;
-    while (isContinue && currentCharacter != '\0')
+    switch (symbol)

```

Comment:
```suggestion
    switch (state)
```
По состоянию нужен switch, а не по символу.
(-1)